examples/gradle-v2/custom-dokka-plugin-example/demo-library/build.gradle.kts

```diff
@@ -0,0 +1,62 @@
+import org.jetbrains.dokka.gradle.engine.plugins.DokkaPluginParametersBaseSpec
+import org.jetbrains.dokka.gradle.internal.InternalDokkaGradlePluginApi
+
+plugins {
+    kotlin("jvm") version "2.0.21"
+    id("org.jetbrains.dokka") version "2.0.20-SNAPSHOT"
+}
+
+dependencies {
+    dokkaPlugin(project(":dokka-plugin-hide-internal-api"))
+}
+
+dokka {
+    moduleName.set("Demo Library")
+
+    pluginsConfiguration {
+
+        // Register the configuration class with Dokka Gradle Plugin
+        registerBinding(HideInternalApiParameters::class, HideInternalApiParameters::class)
+
+        // Configure the custom plugin:
+        register<HideInternalApiParameters>("HideInternalApiPlugin") {
+            // Tell the plugin to hide code annotated with `@HideFromDokka`.
+            annotatedWith.add("demo.HideFromDokka")
+        }
+    }
+}
+
+/**
+ * Define custom parameters for the HideInternalApi Dokka Plugin.
+ *
+ * Using a custom class for defining parameters means Gradle can accurately check for task up-to-date checks.
+ *
+ * If you find you need to re-use the plugin parameters configuration class in multiple buildscripts,
+ * move the class into a shared location, like `buildSrc` (or another included-build for build conventions).
+ * See [Sharing Build Logic between Subprojects](https://docs.gradle.org/8.10/userguide/sharing_build_logic_between_subprojects.html).
+ */
+@OptIn(InternalDokkaGradlePluginApi::class)
```

Comment:
probably we need to drop this annotation from this constructor, it’s not a good idea to advertise usage of `internal` API's in example

Reply:
Yes, at the moment it's a problem that custom Dokka plugins can't provide their own implementation without using the internal `DokkaPluginParametersBaseSpec`. 

However, just creating a subtype of `DokkaPluginParametersBaseSpec` isn't nice. Users have to manually convert the values to JSON, which is annoying and easy to get wrong (like not properly escaping the values).

I think it'd be worth re-implementing `DokkaPluginParametersBuilder` (removed in https://github.com/Kotlin/dokka/pull/3872), but make it better and tested. I want something similar to `CommandLineArgumentProvider`, where users can just create a subclass and define the inputs in the constructor ([Dokka build example](https://github.com/Kotlin/dokka/blob/15986f67452872ac13956d37f7cf090a5c6cfc58/build-logic/src/main/kotlin/dokkabuild/utils/SystemPropertyAdder.kt#L195-L205))

Reply:
> I think it'd be worth re-implementing

Not yet for sure, let's see how much users will need this based on feedback. For now, I think we have enough tasks with higher priority.